### DISCUSSION TITLE
IF Forwarding

### DISCUSSION THREAD

**Discussion by pbtgit** [zephyrproject-rtos/zephyr#50027] [Q&A]
I have two working network interfaces on NRF HW, one using 802.15.4 and one using Ethernet via SLIP.
Now, if i change the SLIP to skip L2 with a "dummy" interface - how can i setup forwarding?
Is it possible to inject/extract L3 packets from the 802.15.4 IF and relay to / from the IF with SLIP?
Any ideas welcome.

Best
Peter

**Comment by jukkar** [comment]
You could try to enable IP routing `CONFIG_NET_ROUTE` and see how far you get. Currently this works only for IPv6 but I suppose this is what you are using with your 802.15.4 interface anyway.
You also need to add the routes yourself using `net_route_add()` and related APIs.

**Reply by pbtgit** [reply]
OK as i understand  if i use CONFIG_NET_ROUTING and then call net_if_ipv6_router_add() i can set a default route which should be good enough for my use-case.

Best
Peter

**Comment by chrta** [comment]
Another way would be to enable `CONFIG_NET_ROUTE` and set IPv6 prefixes for the network interfaces via `net_if_ipv6_prefix_add()`. This information is used if no explicit route is set/found, see https://github.com/zephyrproject-rtos/zephyr/blob/zephyr-v3.1.0/subsys/net/ip/ipv6.c#L351

**Reply by pbtgit** [reply]
Thanks! that seems to be exactly the right kind of feature!

Best
Peter
